### PR TITLE
REST API endpoint for creating and deleting consumers

### PR DESCRIPTION
**Pull request by alexeyvy**
According to https://oat-sa.atlassian.net/browse/TAO-6644

### CODE REVIEW

**Comment by gyszucs** [comment]
@alexeyvy Could you please give some details how to test this?

**Comment by alexeyvy** [comment]
> @alexeyvy Could you please give some details how to test this?

curl -X POST "https://tao.local/taoLti/RestService/index" -H "Authorization: Basic xxxxxxxxxxxxxxx" -H "accept: application/json" -H "Content-Type: application/x-www-form-urlencoded" -d "label=sample&oauth-key=123&oauth-secret=321&oauth-callback-url=43" --insecure

**Comment by alexeyvy** [comment]
Tests have been added

**Comment by alexeyvy** [comment]
>The newly created consumer appears twice on the BackOffice. Do you also have this issue?
Yes, that was an issue. It's fixed now, please try :)

File: models/classes/LtiRestApiService.php [hunk 1]
```diff
@@ -20,18 +20,33 @@
...
+     */
+    protected $consumerClass;
+
+    /**
+     * @throws \common_exception_Error
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->consumerClass = new \core_kernel_classes_Class(ConsumerService::CLASS_URI);
```
[the remark this answers is not in the document]
Comment (alexeyvy) [reply models/classes/LtiRestApiService.php]: done

File: models/classes/LtiRestApiService.php [hunk 2]
```diff
@@ -20,18 +20,29 @@
...
-    protected function getRootClass()
+    /**
+     * @var \core_kernel_classes_Class
+     */
+    protected $consumerClass;
+
+    /**
+     * @throws \common_exception_Error
+     */
+    public function __construct()
```
[the remark this answers is not in the document]
Comment (alexeyvy) [reply models/classes/LtiRestApiService.php]: Done

**Pull request merged**